07_errors/n0npax/memStore.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+// NewMemStore creates new storer for map
+func NewMemStore() MemStore {
+	return MemStore{}
+}
+
+// CreatePuppy creates puppy
+func (m MemStore) CreatePuppy(puppy *Puppy) (int, error) {
+	if puppy.Value < 0 {
+		return -1, Errorf(ErrCodeInvalidInput, "puppy value have to be positive number")
+	}
+	id := len(m)
```

Comment:
this isn't going to work. If you add two puppies, delete the first (id 0) and then add another, it will wipe out the other puppy:
```
func (s *storerSuite) TestCreateDeleteCreate() {
        assert := tassert.New(s.T())
        newPuppy0, newPuppy1 := puppy0(), puppy1()
        s.store.CreatePuppy(&newPuppy0)
        s.store.CreatePuppy(&newPuppy1)
        s.store.DeletePuppy(newPuppy0.ID)
        newerPuppy0 := puppy0()
        s.store.CreatePuppy(&newerPuppy0)
        p, _ := s.store.ReadPuppy(newPuppy1.ID)
        assert.Equal(newPuppy1, *p)
}
```

That test fails for the memstore, but not the others.

Reply:
Awe yes. It's fixed in lab10. I'll port this right now
Sorry

Reply:
fixed